docs/design/submarine-server/architecture.md

```diff
@@ -0,0 +1,126 @@
+	     |    |--server-yarn-runtime
+	     |    |--server-operator
+	     |--submarine-workbench
+	     |    |--workbench-server
+	     |    |--workbench-web
+	     |--submodules
+```
+
+### submarine-client
+The client as the default implements for submarine server RESTful API, it provides all the features which declared in the RESTful API.
+
+### submarine-server
+```
++------+ +--------------------+
+|      | | +------+ +-------+ |
+|      | | | YARN | |  K8s  | |
+|      | | +------+ +-------+ |
+|      | |     submitter      |
+|      | +--------------------+
+| REST | +--------+ +---------+
+|      | | TRANSL | | monitor |
+|      | +--------+ +---------+
+|      | +--------------------+
+|      | |     JobManager     |
++------+ +--------------------+
+```
+The project structure is as follows:
+```
+	|--submarine-server
+	|----server-api
+	|----server-core
+	|----server-submitter
+	|----pom.xml
+	|----README
+```
+The server-sdk will be added in the future. It is the official client for Submarine Server API.
+
+#### REST
+It provides the RESTful API for users, consists of submit job and get job info features and so on. Often used by CLI and workbench etc. The submarine client project should implement all APIs.
+
+#### Submitter
+It provides the submitter interface, and implement it for YARN and K8s. Besides that support extends the other resource management system such as submarine-docker-cluster only implement the interface which declared at server-core project. So that support submit the machine learning job to different cluster resource management system. The project structure is as follows:
+```
+	|--server-submitter
+	|----submitter-yarn
+	|----submitter-yarnservice
+	|----submitter-k8s
+	|----pom.xml
+	|----README
+```
+In here the submitter-yarn implement the TonY Runtime(tony-runtime), and the submitter-yarnservice(yarnservice-runtime) diabled by default until we decide to support the yarn service feature. If we want to support the submarine-docker-cluster (Submarine uses the Raft algorithm to create a docker cluster on the docker runtime environment on multiple servers, providing the most lightweight resource scheduling system for small-scale users.) should create the sub project named submitter-docker and must implement all the interface declared in the server-core project.
+
+#### Monitor
+We need a monitor to track the job life cycle and record the main events and key info in running. If the client register to server, the monitor will called the pusher to push the info to client.
+
+#### Translator
+Convert the job spec to different runtime’s input data. 
+
+#### JobsManager
+JobsManager manages all jobs and schedule the job to the resource management system. It consists of scheduler and timeline etc.
+
+#### Pusher(TODO)
+Pusher provides the API to create collaboration & communication feature. It consists of server and SDK modules. The SDK used by client to keep the connect to server and receive info.
+
+#### DB (optional)
```

Comment:
The DB can be used to do submarine-server failure recovery.

Reply:
Yes, excepts that it can store the job history.